File: tools/generate_test_textures.py
#!/usr/bin/env python3
"""Generate placeholder test textures for PBR material testing"""
import numpy as np
from PIL import Image


def main():
    # Create 256x256 test ORM texture (Red=occlusion, Green=roughness, Blue=metallic)
    orm = np.zeros((256, 256, 3), dtype=np.uint8)
    orm[:, :, 0] = int(0.5 * 255)  # Occlusion (red channel)
    orm[:, :, 1] = int(0.7 * 255)  # Roughness (green channel)
    orm[:, :, 2] = int(0.3 * 255)  # Metallic (blue channel)
    Image.fromarray(orm).save("tests/fixtures/textures/test_orm.png")
    print("✓ Created test_orm.png (ORM: R=0.5, G=0.7, B=0.3)")

    # Create 256x256 emissive texture (red glow pattern - radial gradient)
    emissive = np.zeros((256, 256, 3), dtype=np.uint8)
    y, x = np.ogrid[:256, :256]
    dist = np.sqrt((x - 128)**2 + (y - 128)**2)
    intensity = np.clip(1.0 - dist / 180.0, 0, 1)
    emissive[:, :, 0] = (intensity * 255).astype(np.uint8)  # Red glow
    Image.fromarray(emissive).save("tests/fixtures/textures/test_emissive.png")
    print("✓ Created test_emissive.png (radial red glow pattern)")

    # Create 256x256 depth texture (grayscale depth pattern - circular depression)
    depth = np.zeros((256, 256, 3), dtype=np.uint8)
    depth_value = np.clip(dist / 180.0, 0, 1)
    gray = (depth_value * 255).astype(np.uint8)
    depth[:, :, 0] = gray
    depth[:, :, 1] = gray
    depth[:, :, 2] = gray
    Image.fromarray(depth).save("tests/fixtures/textures/test_depth.png")
    print("✓ Created test_depth.png (grayscale circular depression)")


if __name__ == "__main__":
    main()

File: tools/generate_test_textures.rs
use image::{Rgb, RgbImage};

fn main() {
    // Create 256x256 test ORM texture (Red=0.5 occlusion, Green=0.7 roughness, Blue=0.3 metallic)
    let mut orm = RgbImage::new(256, 256);
    for pixel in orm.pixels_mut() {
        *pixel = Rgb([
            (0.5 * 255.0) as u8, // Occlusion (red channel)
            (0.7 * 255.0) as u8, // Roughness (green channel)
            (0.3 * 255.0) as u8, // Metallic (blue channel)
        ]);
    }
    orm.save("tests/fixtures/textures/test_orm.png").unwrap();
    println!("Created test_orm.png");

    // Create 256x256 emissive texture (red glow pattern - radial gradient)
    let mut emissive = RgbImage::new(256, 256);
    for y in 0..256 {
        for x in 0..256 {
            let dx = (x as f32 - 128.0).abs();
            let dy = (y as f32 - 128.0).abs();
            let dist = (dx * dx + dy * dy).sqrt();
            let intensity = (1.0 - (dist / 180.0).min(1.0)).max(0.0);
            emissive.put_pixel(
                x,
                y,
                Rgb([
                    (intensity * 255.0) as u8, // Red glow
                    0,                         // No green
                    0,                         // No blue
                ]),
            );
        }
    }
    emissive
        .save("tests/fixtures/textures/test_emissive.png")
        .unwrap();
    println!("Created test_emissive.png");

    // Create 256x256 depth texture (grayscale depth pattern - circular depression)
    let mut depth = RgbImage::new(256, 256);
    for y in 0..256 {
        for x in 0..256 {
            let dx = (x as f32 - 128.0).abs();
            let dy = (y as f32 - 128.0).abs();
            let dist = (dx * dx + dy * dy).sqrt();
            // Center is darkest (deepest), edges are lightest (highest)
            let depth_value = (dist / 180.0).min(1.0);
            let gray = (depth_value * 255.0) as u8;
            depth.put_pixel(x, y, Rgb([gray, gray, gray]));
        }
    }
    depth
        .save("tests/fixtures/textures/test_depth.png")
        .unwrap();
    println!("Created test_depth.png");
}
